controllers/taloscontrolplane_controller.go

```diff
@@ -574,6 +567,9 @@ func (r *TalosControlPlaneReconciler) bootControlPlane(ctx context.Context, clus
 	}
 
 	bootstrapConfig := &tcp.Spec.ControlPlaneConfig.ControlPlaneConfig
+	if !reflect.ValueOf(tcp.Spec.ControlPlaneConfig.InitConfig).IsZero() && first {
```

Comment:
I feel `reflect` is a bit of a hack here, there are two ways to fix this:

* add method `IsZero` directly to the type of `InitConfig`
* as this should be Go struct, we can check for zero with `.InitConfig == ControlPlaneConfig{}` (whatever the actual struct type is, I used `ControlPlaneConfig` here)

Reply:
Did some digging on this. InitConfig is a cabptv1.TalosConfigSpec (comes from bootstrap provider). 

So for the first option of IsZero, we'll need to implement that function as part of the bootstrap provider and then pull it in. 

For the second option, we can't easily do that because TalosConfigSpec contains a slice of ConfigPatches, so go barfs on not being able to compare (something like: `cannot compare tcp.Spec.ControlPlaneConfig.InitConfig == ((cabptv1.TalosConfigSpec literal)) (operator == not defined for ...)`). 

Let's chat in the AM about how we wanna solve this.

Reply:
After chatting, merging as-is for now. We can add an IsZero function to CABPT later and update if we want.